docs/en/reference/dql-doctrine-query-language.rst

```diff
@@ -591,23 +591,85 @@ You can also nest several DTO :
         {
             // Bind values to the object properties.
         }
     }
-	
+
 .. code-block:: php
 
     <?php
     $query = $em->createQuery('SELECT NEW CustomerDTO(c.name, e.email, NEW AddressDTO(a.street, a.city, a.zip)) FROM Customer c JOIN c.email e JOIN c.address a');
     $users = $query->getResult(); // array of CustomerDTO
-	
+
 Note that you can only pass scalar expressions or other Data Transfer Objects to the constructor.
 
+If you use your data-transfer objects for multiple queries but not necessarily with the same parameters,
+you can use named arguments or variadic arguments, add ``named`` keyword in your DQL querie before your Dto :
+
+.. code-block:: php
+
+    <?php
+
+    class CustomerDTO
+    {
+        public function __construct(
+                public string|null $name = null,
+                public string|null $email = null,
+                public string|null $city = null,
+                public mixed|null $value = null,
+                public AddressDTO|null $address = null,
+        ) {
+        }
+    }
+
+And then you can select the fields you want in the order you want, and the ORM will try to match argument names with the selected field names :
+
+.. code-block:: php
+
+    <?php
+    $query = $em->createQuery('SELECT NEW NAMED CustomerDTO(a.city, c.name) FROM Customer c JOIN c.address a');
+    $users = $query->getResult(); // array of CustomerDTO
+
+    // CustomerDTO => {name : 'SMITH', email: null, city: 'London', value: null}
+
+ORM will also look column aliases before columns names :
+
+.. code-block:: php
+
+    <?php
+    $query = $em->createQuery('SELECT NEW NAMED CustomerDTO(c.name, CONCAT(a.city, ' ' , a.zip) AS value) FROM Customer c JOIN c.address a');
+    $users = $query->getResult(); // array of CustomerDTO
+
+    // CustomerDTO => {name : 'DOE', email: null, city: null, value: 'New York 10011'}
+
+For aliases a field you can use SQL notation with `AS` keyword, or the PHP named arguments notation :
+
+.. code-block:: php
+
+    <?php
+    $query = $em->createQuery('SELECT NEW NAMED CustomerDTO(name: c.name, value: CONCAT(a.city, ' ' , a.zip)) FROM Customer c JOIN c.address a');
+    $users = $query->getResult(); // array of CustomerDTO
+
+    // CustomerDTO => {name : 'DOE', email: null, city: null, value: 'New York 10011'}
+
+The ``NAMED`` keyword must be added to all dto you want to instantiate :
+
+.. code-block:: php
+
+    <?php
+    $query = $em->createQuery('SELECT NEW NAMED CustomerDTO(name: c.name, address: NEW NAMED AddressDTO(a.street, a.city, a.zip)) FROM Customer c JOIN c.address a');
+    $users = $query->getResult(); // array of CustomerDTO
+
+    // CustomerDTO => {name : 'DOE', email: null, city: null, value: 'New York 10011'}
+
+If two fields have the same name/alias, a RepeatedAliasException is thrown.
+If a field without column name (functions for example) has no alias set, a MissingAliasException is thrown.
```

Comment:
```suggestion
If a field without a column name (when using functions for example) has no alias set, a ``MissingAliasException`` is thrown.
```

Reply:
[When I use a function](https://sqlfiddle.com/postgresql/online-compiler?id=26728015-ebc0-4603-8e34-69c6cc60fa4a) the resulting column does have a name :thinking: